src/game.py

```diff
@@ -9,28 +9,50 @@ def __init__(self, cli=Cli(), board=Board(), players=[Player(token="X"),
         self.cli = cli
         self.board = board
         self.players = players
+        self.exit = False
+        self.safe_exit = False
 
     def start(self):
         self.cli.welcome()
         self.play()
 
     def play(self):
         self.cli.display_board(self.board)
-        while not self.board.full():
+        while not self.game_is_over():
             self.turn()
 
-        self.cli.goodbye()
-    
-    def turn(self, error=False):
-        if error: self.cli.invalid_move()
+    def game_is_over(self):
```

Comment:
Seems strange to have a method called game_is_over where it's worded as a question, yet it has side effects.